Assets/VRTK/Scripts/VRTK_ThirdPersonCamera.cs

```diff
@@ -0,0 +1,113 @@
+        [Tooltip("Because acceleration can sometimes cause simulator sickness, we wanted to try changing the acceleration curve. One easy way to do that is with animation curves. The animation curve can make the acceleration curve slower to curb simulator sickness by having the 'Y' axis grow slower.")]
+        [SerializeField]
+        private AnimationCurve animationCurve;
+        [Tooltip("Determines whether to use animation curve or standard lerp mechanism. Don't enable unless you've created an animation curve")]
+        [SerializeField]
+        private bool useAnimationCurve;
+        [Tooltip("Used to switch camera angles when the grip button is pressed. Remember to assign VRTK_ControllerEvents as a component of your left controller.")]
+        [SerializeField]
+        private VRTK_ControllerEvents leftControllerEvents;
+        [Tooltip("Used to switch camera angles when the grip button is pressed. Remember to assign VRTK_ControllerEvents as a component of your right controller.")]
+        [SerializeField]
+        private VRTK_ControllerEvents rightControllerEvents;
+        private Vector3 targetPreviousPosition;
+        private Transform target;
+        private float animationCurvePercentage = 0;
+        private GameObject currentTransformObject;
+        private int currentIndex = 0;
+        private Transform hmd;
+        // Finds HMD position, then sets the transform of the camera
+        void Start()
+        {
+            hmd = VRTK_DeviceFinder.HeadsetCamera();
+            //if no cameras set, then find the player gameobject.
+            if (targetArray.Length == 0)
+            {
+                Debug.Log("No default cameras discovered, please add some empty game objects to the scene and assign them to the array..");
+            }
+            if(leftControllerEvents == null || rightControllerEvents == null)
+            {
+                Debug.Log("Requires VRTK_ControllerEvents to be attached to controllers and then assigned.");
+            }
+            target = GetTargetPosition(targetArray[0]);
+            transform.position = target.position;
+            leftControllerEvents.GripPressed += ChangeCameraAngle;
+            rightControllerEvents.GripPressed += ChangeCameraAngle;
+        }
+        //Offset target position by where the user was when the view switched, to always get standardized camera view
+        private Transform GetTargetPosition(Transform targetPosition)
+        {
+            if (currentTransformObject != null)
+            {
+                Destroy(currentTransformObject);
+            }
+            currentTransformObject = Instantiate(targetPosition.gameObject, targetPosition.parent) as GameObject;
+            currentTransformObject.transform.position = currentTransformObject.transform.position - transform.InverseTransformPoint(hmd.position);
+            return currentTransformObject.transform;
+
+        }
+      
+        private void Update()
```

Comment:
All MonoBehaviour methods need to be `protected virtual` so people can extend the scripts if required

Also, all methods of the same accessibility should be grouped together, so

all public methods first, then protected methods, then private methods.

I also like to keep the MonoBehaviour methods at the top of that grouping